packages/cli-kit/src/public/node/ui.tsx

```diff
@@ -689,28 +689,40 @@ export const keypress = async (stdin = process.stdin, uiDebugOptions: UIDebugOpt
 
 interface IsTTYOptions {
   stdin?: NodeJS.ReadStream
+  stdout?: NodeJS.WriteStream
   uiDebugOptions?: UIDebugOptions
 }
 
-export function isTTY({stdin = undefined, uiDebugOptions = defaultUIDebugOptions}: IsTTYOptions = {}) {
-  return Boolean(uiDebugOptions.skipTTYCheck || stdin || terminalSupportsRawMode())
+export function isTTY({
+  stdin = undefined,
+  stdout = undefined,
+  uiDebugOptions = defaultUIDebugOptions,
+}: IsTTYOptions = {}) {
+  return (
+    Boolean(uiDebugOptions.skipTTYCheck) ||
+    (Boolean(stdin || terminalSupportsRawMode()) && Boolean(stdout || process.stdout.isTTY))
+  )
 }
```

Comment:
I disagree with the direction here.

Looking over the code generally, we use `terminalSupportsRawMode` as a test of TTY-like behavior. So it makes sense to fix `terminalSupportsRawMode` to look at both stdin and stdout, only returning `true` if `stdin` and `stdout` are both TTY.

And for this function, I'd expect something like:

```suggestion
export function isTTY({
  stdin = undefined,
  stdout = undefined,
  uiDebugOptions = defaultUIDebugOptions,
}: IsTTYOptions = {}) {
  return (
    Boolean(
      uiDebugOptions.skipTTYCheck ||
      (stdin && stdout) ||
      terminalSupportsRawMode(stdin ?? process.stdin, stdout ?? process.stdout)
  )
}
```